traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -479,7 +479,7 @@ func createV40(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, dsV40 t
 		return nil, http.StatusInternalServerError, nil, errors.New("ensuring ds parameters:: " + err.Error())
 	}
 
-	if dnssecEnabled {
+	if dnssecEnabled && !ds.Type.IsAnyMap() {
```

Comment:
Actually, looking at the DNSSEC refresh code here: https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/cdn/dnssecrefresh.go#L195

I think we should refactor that condition into a new method 
```
func (t DSType) UsesDNSSECKeys() bool {
    return t.IsHTTP() || t.IsSteering() || t.IsDNS()
}
```
and use it in both places (i.e. `!t.UsesDNSSECKeys()` here and in dnssecrefresh.go).

The idea being that if we add a new delivery service type, we have to explicitly decide whether or not it uses dnssec keys, rather than getting dnssec keys by default.

Reply:
Changes have been made to the method and refresh code.